spec/models/note_spec.rb

```diff
@@ -16,20 +16,31 @@
     end
   end
 
-  { Grouping: -> { FactoryBot.create(:grouping) },
-    Student: -> { FactoryBot.create(:student) },
-    Assignment: -> { FactoryBot.create(:assignment) } }.each_pair do |type, noteable|
-    context "when #{type} exist" do
-      before do
-        @noteable = noteable.call
-      end
-      it 'returns true for the notable\'s course' do
-        expect(Note.noteables_exist?(@noteable.course.id)).to be true
-      end
-      it 'returns false for a different course' do
-        course = create(:course)
-        expect(Note.noteables_exist?(course.id)).to be false
-      end
+  shared_examples 'testing noteable on different models' do
+    it 'returns true for the notable\'s course' do
+      expect(Note.noteables_exist?(noteable.course.id)).to be true
+    end
+    it 'returns false for a different course' do
+      course = create(:course)
+      expect(Note.noteables_exist?(course.id)).to be false
+    end
+  end
+
+  context 'when Grouping exist' do
+    include_examples 'testing noteable on different models' do
+      let(:noteable) { create(:grouping) }
+    end
+  end
+
+  context 'when Grouping exist' do
```

Comment:
These labels need to be updated (please make sure to be careful when copy and pasting code)

Reply:
Got it!